tools/walletextension/common/constants.go

```diff
@@ -27,28 +27,28 @@ const (
 )
 
 const (
-	PathRoot                            = "/"
-	PathReady                           = "/ready/"
-	PathViewingKeys                     = "/viewingkeys/"
-	PathGenerateViewingKey              = "/generateviewingkey/"
-	PathSubmitViewingKey                = "/submitviewingkey/"
-	PathJoin                            = "/join/"
-	PathAuthenticate                    = "/authenticate/"
-	PathQuery                           = "/query/"
-	PathRevoke                          = "/revoke/"
-	PathObscuroGateway                  = "/"
-	PathHealth                          = "/health/"
-	WSProtocol                          = "ws://"
-	DefaultUser                         = "defaultUser"
-	UserQueryParameter                  = "u"
-	AddressQueryParameter               = "a"
-	MessageFormatRegex                  = `^Register\s(\w+)\sfor\s(\w+)$`
-	MessageUserIDLen                    = 64
-	SignatureLen                        = 65
-	PersonalSignMessagePrefix           = "\x19Ethereum Signed Message:\n%d%s"
-	GetStorageAtUserIDRequestMethodName = "getUserID"
-	SuccessMsg                          = "success"
-	APIVersion1                         = "/v1"
+	PathRoot                         = "/"
+	PathReady                        = "/ready/"
+	PathViewingKeys                  = "/viewingkeys/"
+	PathGenerateViewingKey           = "/generateviewingkey/"
+	PathSubmitViewingKey             = "/submitviewingkey/"
+	PathJoin                         = "/join/"
+	PathAuthenticate                 = "/authenticate/"
+	PathQuery                        = "/query/"
+	PathRevoke                       = "/revoke/"
+	PathObscuroGateway               = "/"
+	PathHealth                       = "/health/"
+	WSProtocol                       = "ws://"
+	DefaultUser                      = "defaultUser"
+	UserQueryParameter               = "u"
+	AddressQueryParameter            = "a"
+	MessageFormatRegex               = `^Register\s(\w+)\sfor\s(\w+)$`
+	MessageUserIDLen                 = 64
+	SignatureLen                     = 65
+	PersonalSignMessagePrefix        = "\x19Ethereum Signed Message:\n%d%s"
+	GetStorageAtUserIDRequestAddress = "0x0000000000000000000000000000000000000000"
```

Comment:
Why change the string to a hex ?

Reply:
Before the change we were using `getUserID` as an address for getting userID with getStorageAt function.
Metamask (and probably also some other tools) check this value and return error if it is not an address, that is why I decided to use GetStorageAt with address `0x0...0` as an address to get userID

Reply:
> Metamask (and probably also some other tools) check this value and return error if it is not an address,

Did you get an error ? Can you post it, I'm wondering bc I didn't get one when implementing this for `listPersonalTransactions`

Reply:
Can you try `walletStore.provider.send` instead ?
Ie. in obscuro scan we use `const personalTxData = await walletStore.provider.send('eth_getStorageAt', ["listPersonalTransactions", requestPayload, null])`

Reply:
The error above is actually recent bug in metamask (https://github.com/MetaMask/metamask-extension/issues/20847).
In another browser I have following error: 
<img width="557" alt="image" src="https://github.com/obscuronet/go-obscuro/assets/4050716/70e56702-b85c-455b-8431-0f4d4fa202f5">

And it appears also if I use:
`const provider = window.ethereum;

    // Define the parameters
    const params = [
        'getUserID',  // Replace with the contract address
        getRandomIntAsString(0, 1000)
    ];

    // Create JSON-RPC payload
    const payload = {
        method: 'eth_getStorageAt',
        params: params,
        id: new Date().getTime(),  // Unique id for the request
        jsonrpc: '2.0'
    };

    // Use `sendAsync` to send the JSON-RPC request
    provider.sendAsync(payload, (error, result) => {
        if (error) {
            console.error('Error:', error);
        } else if (result.error) {
            console.error('Result Error:', result.error);
        } else {
            console.log('Storage value:', result.result);
        }
    });`
    
    
    `.send` method is/will be deprecated soon: https://eips.ethereum.org/EIPS/eip-1193#send-deprecated

Reply:
Perhaps that error can be from the provider usage? Is this using `ethers`? 

Wonder if using something like we did on obscuro_scan would help https://github.com/obscuronet/go-obscuro/blob/f4913937b1f876aebb26eb915134a6a76856372c/tools/obscuroscan_v2/frontend/src/components/MetaMaskConnectButton.vue ?

> .send method is/will be deprecated soon:
> https://eips.ethereum.org/EIPS/eip-1193#send-deprecated

Yeah I think it's deprecated for external providers, one should use sendAsync, but it should be fine to use with ethers.
<img width="860" alt="image" src="https://github.com/obscuronet/go-obscuro/assets/4471462/ad4a29ad-0518-434d-a3fa-a4b7dafb2703">

Reply:
fixed + reverted back to getUserID